### PR TITLE
Attempt to fix danger breaks when files moved

### LINKED ISSUE #553: Danger breaks when files are moved
# Report

## What did you do?

I moved a file: https://github.com/SwiftWeekly/swiftweekly.github.io/pull/89/commits/f51ce4759205e31699f59ba6aebfcc2249f88016

https://github.com/SwiftWeekly/swiftweekly.github.io/pull/89

## What did you expect to happen?

I expected Danger to not attempt to lint the file at its previous location, but rather lint it at its new location.

## What happened instead?

💥. Danger tried to lint the file at its old location, causing the following failure:

```
/home/travis/build/SwiftWeekly/swiftweekly.github.io/vendor/bundle/ruby/2.1.0/gems/danger-prose-2.0.3/lib/danger_plugin.rb:63:in `block (2 levels) in lint_files':  (Danger::DSLError)
[!] Invalid `Dangerfile` file: undefined method `[]' for nil:NilClass. Updating the Danger gem might fix the issue. Your Danger version: 3.2.1, latest Danger version: 3.2.2
 #  from Dangerfile:2
 #  -------------------------------------------
 #  # Look for prose issues
 >  prose.lint_files
 #  
 #  -------------------------------------------
	from /home/travis/build/SwiftWeekly/swiftweekly.github.io/vendor/bundle/ruby/2.1.0/gems/danger-prose-2.0.3/lib/danger_plugin.rb:63:in `select'
	from /home/travis/build/SwiftWeekly/swiftweekly.github.io/vendor/bundle/ruby/2.1.0/gems/danger-prose-2.0.3/lib/danger_plugin.rb:63:in `block in lint_files'
	from /home/travis/build/SwiftWeekly/swiftweekly.github.io/vendor/bundle/ruby/2.1.0/gems/danger-prose-2.0.3/lib/danger_plugin.rb:179:in `with_proselint_disabled'
	from /home/travis/build/SwiftWeekly/swiftweekly.github.io/vendor/bundle/ruby/2.1.0/gems/danger-prose-2.0.3/lib/danger_plugin.rb:60:in `lint_files'
	from Dangerfile:2:in `block in parse'
	from /home/travis/build/SwiftWeekly/swiftweekly.github.io/vendor/bundle/ruby/2.1.0/gems/danger-3.2.1/lib/danger/danger_core/dangerfile.rb:194:in `eval'
	from /home/travis/build/SwiftWeekly/swiftweekly.github.io/vendor/bundle/ruby/2.1.0/gems/danger-3.2.1/lib/danger/danger_core/dangerfile.rb:194:in `block in parse'
	from /home/travis/build/SwiftWeekly/swiftweekly.github.io/vendor/bundle/ruby/2.1.0/gems/danger-3.2.1/lib/danger/danger_core/dangerfile.rb:190:in `instance_eval'
	from /home/travis/build/SwiftWeekly/swiftweekly.github.io/vendor/bundle/ruby/2.1.0/gems/danger-3.2.1/lib/danger/danger_core/dangerfile.rb:190:in `parse'
	from /home/travis/build/SwiftWeekly/swiftweekly.github.io/vendor/bundle/ruby/2.1.0/gems/danger-3.2.1/lib/danger/danger_core/executor.rb:46:in `run'
	from /home/travis/build/SwiftWeekly/swiftweekly.github.io/vendor/bundle/ruby/2.1.0/gems/danger-3.2.1/lib/danger/commands/runner.rb:56:in `run'
	from /home/travis/build/SwiftWeekly/swiftweekly.github.io/vendor/bundle/ruby/2.1.0/gems/claide-1.0.0/lib/claide/command.rb:334:in `run'
	from /home/travis/build/SwiftWeekly/swiftweekly.github.io/vendor/bundle/ruby/2.1.0/gems/danger-3.2.1/bin/danger:5:in `<top (required)>'
	from /home/travis/build/SwiftWeekly/swiftweekly.github.io/vendor/bundle/ruby/2.1.0/bin/danger:23:in `load'
	from /home/travis/build/SwiftWeekly/swiftweekly.github.io/vendor/bundle/ruby/2.1.0/bin/danger:23:in `<main>'
```

## Your Environment

* Which CI are you running on?

Travis, though I don't think that's related. Here's the failing build: https://travis-ci.org/SwiftWeekly/swiftweekly.github.io/builds/160181293

* Are you running the latest version of Danger?

Running 3.2.1, but looking at the diff between that and `master`, nothing that's landed since would have fixed this.

### PR DESCRIPTION
**Pull request by JuanitoFatas**
This Pull Request fixes #553 💪

#### Main changes

- Access Octokit's sawyer (json-like) object via string keys and fix specs
- Add [a spec](https://github.com/danger/danger/blob/a6dff6f8c156c24b03a053b424ec46e2e1b065dc/spec/lib/danger/danger_core/executor_spec.rb) that runs the whole thing for [executor.rb](https://github.com/danger/danger/blob/5708273eb5feb20c208d905b9d01195c10420c9a/lib/danger/danger_core/executor.rb)

#### Reasoning

I think access [Octokit](https://github.com/octokit/octokit.rb)'s [sawyer object](https://github.com/lostisland/sawyer) (`self.pr_json`, etc) via keys of `String` is more reliable (I have been bitten from building https://www.deppbot.com). But I have no evidence to support this... 😓  From my experience, symbolized keys may result in `nil` value, hence our base & head commits are `nil` => no diff 😱 .

--

- [x] HOWTO put a real git repo in `spec/fixtures` => Fake it, don't put a real git repo 733c7ff

### CODE REVIEW

**Comment by JuanitoFatas** [comment]
@orta It seems this Pull Request fixes the problem of moving files, see a6dff6f. But please don't merge, after you approved, I will remove a6dff6f, then we can merge. 🙇

File: spec/lib/danger/danger_core/executor_spec.rb [hunk 1]
```diff
@@ -1,35 +1,129 @@
...
+    ENV["DANGER_GITHUB_API_TOKEN"] = "1234567890" * 4 # octokit token is of size 40
 
-      described_class.new({}).post_results(danger_file, danger_id)
+    yield
+  ensure
+    ENV["HAS_JOSH_K_SEAL_OF_APPROVAL"] = approval
+    ENV["TRAVIS_PULL_REQUEST"] = pr
+    ENV["TRAVIS_REPO_SLUG"] = slug
+    api_token = api_token
+  end
```
Comment (orta) [inline spec/lib/danger/danger_core/executor_spec.rb]: nice touch

**Comment by orta** [comment]
This is good to go after a quick rebase

**Comment by orta** [comment]
Alright - lets do this 3.3.1

**Pull request merged**